tests/test_signed_payload.py

```diff
@@ -3,74 +3,120 @@
+            "incorrectAuthorizationHeader": "Signature sig"
         }
 
-        signature = pk.sign_msg(json.dumps(payload))
-        print json.dumps(payload)
-        print pk.public_key.to_checksum_address().lower()
-        print base64.b64encode(signature.to_bytes())
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
 
-    def test_signed_payload(self):
-        payload = {
-            "service_proposal": {
-                "id": 1,
-                "format": "service-proposal/v1",
-                "provider_id": "node1",
-            }
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'missing Authorization in request header')
+
+    def test_incorrect_authorization_header_value_format(self):
+        headers = {
+            "Authorization": "Signature"
         }
 
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
+
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'invalid Authorization header value provided, correct format: Signature <signature_base64_encoded>')
+
+    def test_incorrect_authorization_type(self):
         headers = {
-            "identity": "0x1a642f0e3c3af545e7acbd38b07251b3990914f1",
-            "Authorization": "Signature Da1mAwK5abmXQCNsCE+YjsZbR9jTyEKqdrjxxMKwNzwr2NFnM35UiVQJWcg8rgL+X2PR60LoIUMlGU9OPaSoZwE="
+            "Authorization": "incorrectType sig"
         }
 
         re = self.client.post(
-            '/v1/signed_payload',
-            data=json.dumps(payload),
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
             headers=headers
         )
 
-        self.assertEqual({}, re.json)
-        self.assertEqual(200, re.status_code)
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'authentication type have to be Signature')
 
-    def test_incorrectly_signed_payload(self):
-        payload = {}
+    def test_empty_authorization_header_value(self):
+        headers = {
+            "Authorization": "Signature "
+        }
+
+        re = self.client.post(
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
+            headers=headers
+        )
 
+        self.assertEqual(401, re.status_code)
+        self.assertEqual(re.json['error'], 'signature was not provided')
+
+    def test_signature_not_base64_encoded(self):
         headers = {
-            "identity": "0x0000000000000000000000000000000000000001",
-            "Authorization": "Signature Da1mAwK5abmXQCNsCE+YjsZbR9jTyEKqdrjxxMKwNzwr2NFnM35UiVQJWcg8rgL+X2PR60LoIUMlGU9OPaSoZwE="
+            "Authorization": "Signature not_base_64"
         }
 
         re = self.client.post(
-            '/v1/signed_payload',
+            '/v1/test_signed_payload',
+            data=json.dumps(''),
```

Comment:
From this line I realised that `@validate_json` doesn't validate whether request is json or not.
It seems that all `@validate_json` does is checks whether content-type is correct.
This is not related to this PR, but is this behaviour intentional? Shouldn't `@validate_json` actually validate that request is a json?

Reply:
It turns out that string `""` is a valid json, like any string: `"hello"`! :O
Ok, nevermind this comment.